pxr/usd/sdf/accessorHelpers.h

```diff
@@ -173,6 +173,13 @@ SDF_ACCESSOR_CLASS::name_(                                                     \
 
 // Convenience macros to provide common combinations of value accessors
 
```

Comment:
Okay, so I think we could make this more complete by adding to the conditional by making it 
std::Is_arithmetic<T>::value || std::is_enum<T>::value
All the places I see that use the SDF_DEFINE_TYPED_GET_SET macro are specifically to for enum values which leads me to believe enums were a hole in the original boost call_traits.
However, I'm okay with leaving this as is in your change and following a followup bug to add is_enum and replace the SDF_DEFINED_TYPED_GET_SET usages